cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -196,38 +517,683 @@ impl BuildEnv for BuildEnvNix {
+                .unwrap(),
+            "store path should be valid before building"
+        );
+
+        let result = buildenv.realise_flakes(&locked_package);
+        assert!(result.is_ok(), "failed to skip building flake");
+    }
+}
+
+#[cfg(test)]
+mod realise_store_path_tests {
+    use super::*;
+    use crate::models::manifest::DEFAULT_PRIORITY;
+
+    fn mock_store_path(valid: bool) -> LockedPackageStorePath {
+        LockedPackageStorePath {
+            install_id: "mock".to_string(),
+            store_path: if valid {
+                env!("GIT_PKG").to_string()
+            } else {
+                "/nix/store/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-invalid".to_string()
+            },
+            system: env!("NIX_TARGET_SYSTEM").to_string(),
+            priority: DEFAULT_PRIORITY,
+        }
+    }
+
+    #[test]
+    fn store_path_build_success_if_valid() {
+        let buildenv = BuildEnvNix;
+        let locked = mock_store_path(true);
+
+        // show that the store path is valid
+        assert!(buildenv.check_store_path([&locked.store_path]).unwrap());
+
+        buildenv
+            .realise_store_path(&locked)
+            .expect("an existing store path should realise");
+    }
+
+    #[test]
+    fn store_path_build_failure_if_invalid() {
+        let buildenv = BuildEnvNix;
+        let locked = mock_store_path(false);
+
+        // show that the store path is invalid
+        assert!(!buildenv.check_store_path([&locked.store_path]).unwrap());
+
+        let result = buildenv
+            .realise_store_path(&locked)
+            .expect_err("invalid store path should fail to realise");
+        assert!(matches!(result, BuildEnvError::Realise2 { .. }));
+    }
+}
+
+#[cfg(test)]
+mod buildenv_tests {
+    use std::collections::HashSet;
+    use std::os::unix::fs::PermissionsExt;
+
+    use regex::Regex;
+
+    use super::*;
+    use crate::providers::catalog::{GENERATED_DATA, MANUALLY_GENERATED};
+
+    trait PathExt {
+        fn is_executable_file(&self) -> bool;
+    }
+
+    impl PathExt for Path {
+        fn is_executable_file(&self) -> bool {
+            self.is_file() && self.metadata().unwrap().permissions().mode() & 0o111 != 0
+        }
+    }
+
+    static BUILDENV_RESULT_SIMPLE_PACKAGE: LazyLock<BuildEnvOutputs> = LazyLock::new(|| {
+        let buildenv = BuildEnvNix;
+        let lockfile_path = GENERATED_DATA.join("envs/hello/manifest.lock");
+        buildenv.build(&lockfile_path, None).unwrap()
+    });
+
+    #[test]
+    fn build_contains_binaries() {
+        let result = &*BUILDENV_RESULT_SIMPLE_PACKAGE;
+        let runtime = &result.runtime;
+        assert!(runtime.join("bin/hello").exists());
+        assert!(runtime.join("bin/hello").is_executable_file());
+
+        let develop = result.develop.as_ref();
+        assert!(develop.join("bin/hello").exists());
+        assert!(develop.join("bin/hello").is_executable_file());
+    }
```

Comment:
(nb) We're no longer testing this against a locked flake. Is that intentional, because we have unit test coverage for realising flakes and the rest of the mechanism is the same?

https://github.com/flox/flox/blob/6916afa816b6b59725a735f59f70b7cb643036f0/cli/tests/buildenv.bats#L65-L75

Reply:
essentially, yes
the only thing testing a locked flake would be test against would be whether buildenv.nix correctly identifies a flake installable, especially while we don't yet honor the `outputs_to_install` (which for flakes uses kebab-case 🫠)